### PR TITLE
Add `GITHUB_ACTIONS_COMMAND_ARGS` environment variable to define additional arguments for `actionlint` command

### LINKED ISSUE #1852: Allow to specify command line argument of actionlint
**Is your feature request related to a problem? Please describe.**

[actionlint](https://github.com/rhysd/actionlint) provides `-ignore` option to ignore some errors. This is necessary to ignore some false positives or errors which a user is not interested in. However, super-linter currently does not provide a way to pass command line options to actionlint.

Related issue: https://github.com/rhysd/actionlint/issues/26

**Describe the solution you'd like**

I'd like to have the way to ignore some errors from actionlint. It'd be nice to have some environment variable for it.

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by rhysd**
Fixes #1852

## Proposed Changes

1. Define `GITHUB_ACTIONS_COMMAND_ARGS` environment variable
2. Pass the environment variable as argument of `actionlint`

### Author/Contributor
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

---

This is a revised PR of #1853.

The context of this change is that `actionlint` provides some configuration via only command line arguments (for example, ignore some errors). But there is no way to add arguments in Super-Linter. This PR adds a way to define additional arguments via the environment variable.

#1853 was closed in favor of #1667. But #1667 was closed as stale. So the issue #1852 was left unresolved.

### CODE REVIEW

**Comment by rhysd** [comment]
Ci seemed to fail at `TYPESCRIPT_STANDARD` linter, unrelated to this PR.

```
2022-10-16 12:05:59 [ERROR]   ERRORS FOUND in TYPESCRIPT_STANDARD:[1]
```

File: lib/linter.sh [hunk 1]
```diff
@@ -891,7 +891,11 @@ LINTER_COMMANDS_ARRAY['DART']="dartanalyzer --fatal-infos --fatal-warnings --opt
 LINTER_COMMANDS_ARRAY['DOCKERFILE_HADOLINT']="hadolint -c ${DOCKERFILE_HADOLINT_LINTER_RULES}"
 LINTER_COMMANDS_ARRAY['EDITORCONFIG']="editorconfig-checker -config ${EDITORCONFIG_LINTER_RULES}"
 LINTER_COMMANDS_ARRAY['ENV']="dotenv-linter"
-LINTER_COMMANDS_ARRAY['GITHUB_ACTIONS']="actionlint -config-file ${GITHUB_ACTIONS_LINTER_RULES}"
+if [ "${GITHUB_ACTIONS_COMMAND_ARGS}" = "null" ] || [ -z "${GITHUB_ACTIONS_COMMAND_ARGS}" ]; then
```
Comment (ferrarimarco) [inline lib/linter.sh:894]: Set the default value as we're doing with the `KUBERNETES_KUBEVAL_OPTIONS` variable

**Review by ferrarimarco — CHANGES_REQUESTED** [review]

**Comment by ferrarimarco** [comment]
Seems so:

```
2022-10-16 12:05:56 [DEBUG]   Running LintCodebase. FILE_TYPE: TYPESCRIPT_STANDARD. Linter name: ts-standard, linter command: ts-standard --parser @typescript-eslint/parser --plugin @typescript-eslint/eslint-plugin --project /tmp/lint/.github/linters/tsconfig.json --env browser --env es6 --env jest, TEST_CASE_RUN: true, FILTER_REGEX_INCLUDE: , FILTER_REGEX_EXCLUDE:  files to lint: /tmp/lint/.automation/test/jscpd/typescript_good_1 copy.ts /tmp/lint/.automation/test/jscpd/typescript_good_1.ts /tmp/lint/.automation/test/typescript_es/typescript_bad_1.ts /tmp/lint/.automation/test/typescript_es/typescript_good_1.ts /tmp/lint/.automation/test/typescript_standard/typescript_bad_1.ts /tmp/lint/.automation/test/typescript_standard/typescript_good_1.ts
2022-10-16 12:05:56 [INFO]   Testing Codebase [TYPESCRIPT_STANDARD] files...
2022-10-16 12:05:56 [INFO]   ----------------------------------------------
2022-10-16 12:05:56 [INFO]   ----------------------------------------------
2022-10-16 12:05:56 [DEBUG]   Linting FILE: /tmp/lint/.automation/test/jscpd/typescript_good_1 copy.ts
2022-10-16 12:05:56 [DEBUG]   File: /tmp/lint/.automation/test/jscpd/typescript_good_1 copy.ts, FILE_NAME: typescript_good_1 copy.ts, DIR_NAME:/tmp/lint/.automation/test/jscpd, FILE_STATUS: good, INDIVIDUAL_TEST_FOLDER: typescript_standard, TEST_CASE_DIRECTORY: .automation/test/typescript_standard
2022-10-16 12:05:56 [DEBUG]   TYPESCRIPT_STANDARD expects to lint individual files. Updated TEST_CASE_DIRECTORY to: .automation/test/typescript_standard/
2022-10-16 12:05:56 [DEBUG]   Skipping /tmp/lint/.automation/test/jscpd/typescript_good_1 copy.ts because it's not in the test case directory for TYPESCRIPT_STANDARD...
2022-10-16 12:05:56 [DEBUG]   Linting FILE: /tmp/lint/.automation/test/jscpd/typescript_good_1.ts
2022-10-16 12:05:56 [DEBUG]   File: /tmp/lint/.automation/test/jscpd/typescript_good_1.ts, FILE_NAME: typescript_good_1.ts, DIR_NAME:/tmp/lint/.automation/test/jscpd, FILE_STATUS: good, INDIVIDUAL_TEST_FOLDER: typescript_standard, TEST_CASE_DIRECTORY: .automation/test/typescript_standard
2022-10-16 12:05:56 [DEBUG]   TYPESCRIPT_STANDARD expects to lint individual files. Updated TEST_CASE_DIRECTORY to: .automation/test/typescript_standard/
2022-10-16 12:05:56 [DEBUG]   Skipping /tmp/lint/.automation/test/jscpd/typescript_good_1.ts because it's not in the test case directory for TYPESCRIPT_STANDARD...
2022-10-16 12:05:56 [DEBUG]   Linting FILE: /tmp/lint/.automation/test/typescript_es/typescript_bad_1.ts
2022-10-16 12:05:56 [DEBUG]   File: /tmp/lint/.automation/test/typescript_es/typescript_bad_1.ts, FILE_NAME: typescript_bad_1.ts, DIR_NAME:/tmp/lint/.automation/test/typescript_es, FILE_STATUS: bad, INDIVIDUAL_TEST_FOLDER: typescript_standard, TEST_CASE_DIRECTORY: .automation/test/typescript_standard
2022-10-16 12:05:56 [DEBUG]   TYPESCRIPT_STANDARD expects to lint individual files. Updated TEST_CASE_DIRECTORY to: .automation/test/typescript_standard/
2022-10-16 12:05:56 [DEBUG]   Skipping /tmp/lint/.automation/test/typescript_es/typescript_bad_1.ts because it's not in the test case directory for TYPESCRIPT_STANDARD...
2022-10-16 12:05:56 [DEBUG]   Linting FILE: /tmp/lint/.automation/test/typescript_es/typescript_good_1.ts
2022-10-16 12:05:56 [DEBUG]   File: /tmp/lint/.automation/test/typescript_es/typescript_good_1.ts, FILE_NAME: typescript_good_1.ts, DIR_NAME:/tmp/lint/.automation/test/typescript_es, FILE_STATUS: good, INDIVIDUAL_TEST_FOLDER: typescript_standard, TEST_CASE_DIRECTORY: .automation/test/typescript_standard
2022-10-16 12:05:56 [DEBUG]   TYPESCRIPT_STANDARD expects to lint individual files. Updated TEST_CASE_DIRECTORY to: .automation/test/typescript_standard/
2022-10-16 12:05:56 [DEBUG]   Skipping /tmp/lint/.automation/test/typescript_es/typescript_good_1.ts because it's not in the test case directory for TYPESCRIPT_STANDARD...
2022-10-16 12:05:56 [DEBUG]   Linting FILE: /tmp/lint/.automation/test/typescript_standard/typescript_bad_1.ts
2022-10-16 12:05:56 [DEBUG]   File: /tmp/lint/.automation/test/typescript_standard/typescript_bad_1.ts, FILE_NAME: typescript_bad_1.ts, DIR_NAME:/tmp/lint/.automation/test/typescript_standard, FILE_STATUS: bad, INDIVIDUAL_TEST_FOLDER: typescript_standard, TEST_CASE_DIRECTORY: .automation/test/typescript_standard
2022-10-16 12:05:56 [DEBUG]   TYPESCRIPT_STANDARD expects to lint individual files. Updated TEST_CASE_DIRECTORY to: .automation/test/typescript_standard/
2022-10-16 12:05:56 [INFO]   ---------------------------
2022-10-16 12:05:56 [INFO]   File:[/tmp/lint/.automation/test/typescript_standard/typescript_bad_1.ts]
2022-10-16 12:05:56 [INFO]    - File:[typescript_bad_1.ts] failed test case (Error code: 1) with [ts-standard] successfully
2022-10-16 12:05:57 [DEBUG]   Error code: 1. Command output:
------
Unable to locate the project file. A project file (tsconfig.json or tsconfig.eslint.json) is required in order to use ts-standard.
------
2022-10-16 12:05:57 [DEBUG]   Linting FILE: /tmp/lint/.automation/test/typescript_standard/typescript_good_1.ts
2022-10-16 12:05:57 [DEBUG]   File: /tmp/lint/.automation/test/typescript_standard/typescript_good_1.ts, FILE_NAME: typescript_good_1.ts, DIR_NAME:/tmp/lint/.automation/test/typescript_standard, FILE_STATUS: good, INDIVIDUAL_TEST_FOLDER: typescript_standard, TEST_CASE_DIRECTORY: .automation/test/typescript_standard
2022-10-16 12:05:57 [DEBUG]   TYPESCRIPT_STANDARD expects to lint individual files. Updated TEST_CASE_DIRECTORY to: .automation/test/typescript_standard/
2022-10-16 12:05:57 [INFO]   ---------------------------
2022-10-16 12:05:57 [INFO]   File:[/tmp/lint/.automation/test/typescript_standard/typescript_good_1.ts]
2022-10-16 12:05:57 [DEBUG]   Found errors. Error code: 1, File type: TYPESCRIPT_STANDARD, Error on missing exec bit: true
2022-10-16 12:05:57 [ERROR]   Found errors in [ts-standard] linter!
2022-10-16 12:05:57 [ERROR]   Error code: 1. Command output:
------
Unable to locate the project file. A project file (tsconfig.json or tsconfig.eslint.json) is required in order to use ts-standard.
------
2022-10-16 12:05:57 [DEBUG]   Error code: 1. Command output:
------
Unable to locate the project file. A project file (tsconfig.json or tsconfig.eslint.json) is required in order to use ts-standard.
------
```

File: lib/linter.sh (hunk 1 above)
Comment (rhysd) [reply lib/linter.sh]: Thank you for the point. I initialized the variable with the default value `null` at 05f6de33deaae3cb31cc328686aa806f5860f030.

**Review by ferrarimarco — APPROVED** [review]

**Comment by lindluni** [comment]
@rhysd 

Thank you for the enhancement. We just fixed our dependency chain after a few months of it being broken due to an upstream dependency. Can you rebase your branch on the latest changes to pick up the vulnerability fix that is flagged here.

**Comment by lindluni** [comment]
Added here: https://github.com/github/super-linter/pull/3772

**Pull request closed without merging**